aqt/installer.py

```diff
@@ -100,6 +100,10 @@ def run(self, arg=None) -> int:
                 self.show_help()
             return 1
         except Exception as e:
+            # For testing: Allow blocked socket errors to cause failed tests
+            if e.__class__.__name__ in ("SocketBlockedError", "SocketConnectBlockedError"):
+                raise e from e
+
```

Comment:
I think this was a really bad idea, and I'm going to change it.

The problem that this is trying to solve is: During tests, how do we differentiate between an exception that was caught properly (via the `except AqtException` clause above) vs. an exception that was unexpected? If I write a test that checks the return code (1 or 0), and I expect the return code to be 1, I have no idea which exception handler caught the exception unless I read from stderr. What I really need is an easy way to check whether or not I triggered the `except Exception` branch, without reading stderr.

The straightforward solution would be something like this:
```python
try:
  args.func(args)
  # success
  return 0
except AqtException:
  # expected error
  return 1
except Exception:
  # catastrophic unexpected error
  return 2  # or some other number
```
This way, we can write write tests that can use the return value to check for unexpected events, rather than reading stderr.

The only problem with this is that I'm not sure what error code to use for "catastrophic unexpected error". I don't need this number to be 2, but I need it to be something other than 0 or 1.

Reply:
For bash script, 2 means misuse of shell-builtin.
https://tldp.org/LDP/abs/html/exitcodes.html

There seems to be a range which has a special meaning, 126-165, 255.
It may be better to use a number other than these.

Reply:
How do you feel about 254?

Reply:
254 looks OK